docs/developer/architecture/core/uisettings-service.asciidoc

```diff
@@ -38,3 +92,66 @@ export class MyPlugin implements Plugin {
 }
 
 ----
+
+Registeration only requires a schema for validation on read and write, all {kib-repo}blob/{branch}/docs/development/core/server/kibana-plugin-core-server.uisettingsparams.md[other parameters] are optional.
+
+See:
+
+- {kib-repo}blob/{branch}/docs/development/core/server/kibana-plugin-core-server.uisettingsservicesetup.register.md[UI settings service Setup API docs]
+
+==== Migrations
+Because `uiSettings` are persisted in the `config` saved object, changing or removing a setting requires a migration of the whole saved object. For example, if we wanted to remove the `custom` setting in the example above and then also rename `my_setting:fourtyTwo` to `my_other_setting:fourtyTwo`, we'd have two migration entries, one for each change targeting the required versions:
```

Comment:
As I'm reading this I'm wondering: do we have a deprecation policy/service for the UI Settings?

I'm thinking about the use case when a UI setting is _globally_ overridden in `kibana.yml`, and then it's removed/renamed. Would it crash Kibana? How do we notify the user about the new name?

Reply:
Plugins can leverage the optional [`deprecation`](https://github.com/elastic/kibana/blob/main/docs/development/core/public/kibana-plugin-core-public.uisettingsparams.deprecation.md) parameter on registration for handling deprecation notices and renames etc. The deprecation warnings are rendered in Advanced Settings and should also be mirrored in the `Configure Kibana` guide (as we do in [the docs for 7.17](https://www.elastic.co/guide/en/kibana/7.17/settings.html)).

Renames and removals are all handled through `config` migrations, which take care of settings that have been changed in Advanced settings.

Overrides are validated at start-up and Kibana will throw a configuration error if there's an issue.

Reply:
@TinaHeiligers On this topic, maybe we should explicitly call out that 3rd party plugins _cannot_ currently register migrations for their advanced settings? (i.e. at the moment, if a 3rd party plugin registers an advanced setting, it's essentially permanent and cannot be fixed without a manual migration, as we can't merge their migrations into core)

Reply:
One other idea 🙂  maybe we can link to the docs on registering migrations from here.

Reply:
Thank you both!